elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/query/SQLColumnProjection.java

```diff
@@ -81,9 +84,23 @@ default boolean canNest(Queryable source, MetaDataStore metaDataStore) {
         boolean canNest = verifier.verify(sql);
         if (! canNest) {
             LOGGER.debug("Unable to nest {} because {}", this.getName(), verifier.getLastError());
+
+            return false;
         }
 
-        return canNest;
+        List<Reference> references = new ExpressionParser(metaDataStore).parse(source, this);
+
+        //Search for any join expression that contains $$column.  If found, we cannot nest
+        //because rewriting the SQL in the outer expression will lose the context of the calling $$column.
+        return references.stream()
+                .map(reference -> reference.accept(
+                        new ReferenceExtractor<JoinReference>(JoinReference.class, metaDataStore, source)))
+                .flatMap(Set::stream)
+                .map(reference -> reference.accept(
+                        new ReferenceExtractor<ColumnArgReference>(ColumnArgReference.class, metaDataStore, source)))
```

Comment:
should the source here be `null`? If it's null, it will start from a join expression and check everything inside the reference tree of the join expression.

Reply:
Yes